File: Cpp/1.两数之和.cpp
/*
 * @lc app=leetcode.cn id=1 lang=cpp
 *
 * [1] 两数之和
 */

// @lc code=start
#include<vector>
#include<algorithm>
using namespace std;

class Solution {

public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int i,j;
        for(i=0;i<nums.size()-1;i++){
            for(j=i+1;j<nums.size();j++){
                if(nums[i]+nums[j]==target){
                   return {i,j};
                }
            }
        }
        return {};
    }
};

// @lc code=end

File: Python/1.两数之和.py
#
# @lc app=leetcode.cn id=1 lang=python
#
# [1] 两数之和
#

# @lc code=start
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        dic=dict()
        for i, value in enumerate(nums):
            if (target-value) in dic.keys():
                return [dic[target-value] ,i]
            else:
                dic[value]=i
        return []


# @lc code=end

